File: acm_archive/6138.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

bool in_vector(int obj_angle, vector<int>& permitted_hours) {
	for(int i = 0; i < permitted_hours.size(); i++) {
		if (obj_angle == permitted_hours[i])
			return true;
	}
	return false;
}

int main() {
	int angle;
	vector<int> permitted_hours;
	for(int i = 0; i < 720; i++) {
		int minutes = i % 60;
		int hours = i / 12;
		int current_angle = abs(minutes - hours);
		permitted_hours.push_back(current_angle *6);
	}
	while(cin>>angle) {
		if (in_vector(angle, permitted_hours)) {
			cout<<"Y\n";
		} else {
			cout<<"N\n";
		}
	}
	return 0;
}

File: acm_archive/6138.py
from sys import stdin, stdout

def rdl():
	return stdin.readline().replace('\n', '')

def rdli():
	return int(rdl())

def main():
	obj_angle = rdli()
	permitted_minutes = []
	for i in range(0, 60):
		minutes = i % 60
		hours = i // 12
		current_angle = abs(minutes - hours)
		permitted_minutes.append(current_angle*6)
	while (obj_angle != ''):
		obj_angle = int(obj_angle)
		if (obj_angle >= 90):
			obj_angle -= 90
		answer = None
		if (obj_angle in permitted_minutes):
			answer = 'Y'
		else:
			answer = 'N'
		stdout.write('{}\n'.format(answer))
		obj_angle = rdl()

if __name__ == '__main__':
	main()